src/app/shared/comment-editor/comment-editor.component.ts

```diff
@@ -280,4 +295,81 @@ export class CommentEditorComponent implements OnInit {
       this.dropArea.nativeElement.classList.remove('highlight-drag-box-disabled');
     }
   }
+
+  private isControlKeyPressed(event) {
+    return event.ctrlKey || event.metaKey;
+  }
+
+  private insertOrRemoveCharsFromHighlightedText(char) {
+    const selectionStart = this.commentTextArea.nativeElement.selectionStart;
+    const selectionEnd = this.commentTextArea.nativeElement.selectionEnd;
+    const currentText = this.commentField.value;
+    const highlightedText = currentText.slice(selectionStart, selectionEnd);
+
+    if (this.hasCharsBeforeAndAfterHighlight(selectionStart, selectionEnd, currentText, char)) {
+      this.removeCharsBeforeAndAfterHighlightedText(selectionStart, selectionEnd, currentText, highlightedText, char);
+    } else if (this.hasCharsInTrimmedHighlight(highlightedText, char)) {
+      this.removeCharsFromHighlightedText(selectionStart, selectionEnd, currentText, highlightedText, char);
+    } else {
+      this.addCharsToHighlightedText(selectionStart, selectionEnd, currentText, highlightedText, char);
```

Comment:
Let's be consistent with naming, since used `insertOrRemoveCharsFromHighlightedText` then here should `insertChars...`

or change both to `add` that's fine to

Reply:
ah good point, i've changed it to `insertChars...`